server/src/main/java/org/elasticsearch/index/engine/LuceneSyntheticSourceChangesSnapshot.java

```diff
@@ -83,7 +84,9 @@ public LuceneSyntheticSourceChangesSnapshot(
         this.maxMemorySizeInBytes = maxMemorySizeInBytes > 0 ? maxMemorySizeInBytes : 1;
         this.sourceLoader = mapperService.mappingLookup().newSourceLoader(null, SourceFieldMetrics.NOOP);
         Set<String> storedFields = sourceLoader.requiredStoredFields();
-        this.storedFieldLoader = StoredFieldLoader.create(false, storedFields);
+        String codec = EngineConfig.INDEX_CODEC_SETTING.get(mapperService.getIndexSettings().getSettings());
+        boolean shouldForceSequentialReader = CodecService.BEST_COMPRESSION_CODEC.equals(codec);
+        this.storedFieldLoader = StoredFieldLoader.create(false, storedFields, shouldForceSequentialReader);
```

Comment:
Maybe we should use always force a sequential reader. I think no matter the stored field format, it has benefits given how `LuceneSyntheticSourceChangesSnapshot` access stored fields.

Reply:
I am not entirely sure. Does this not depend upon how jumpy the data is, which with index sorting could be very jumpy - and whether there is more than one stored field per doc (not sure that is always true?)?

Reply:
Right, this is why then changed to logic not look at index.codec setting anymore.